### DISCUSSION TITLE
Remote environment and local debugging

### DISCUSSION THREAD

**Discussion by Terrorboy** [spatie/ray#119] [Q&A] [answered]
HI?
I am very interested in this program.

https://github.com/Terrorboy/request_viewer
Likewise, I am using a similar function on a private server.

Most of the time I work by connecting to a remote server.

When I think about the way the program runs, it seems that ray creates a local server and receives the data sent from the project.

In the case of me working on a remote server, the remote server debugging is done on the local server (ex. xdebug). Can I replace this with ray?

It seems difficult to receive data from remote server from ray local server (unless ngrok is included...)

Do you have any future support plans?

**Comment by freekmurze** [comment, marked as answer]
We are actively working on this , and will add basic support for this in the next few weeks.

**Reply by Terrorboy** [reply]
Thank you.
I bought it with confidence!!

**Reply by sunscreem** [reply]
@freekmurze Just and FYI - I use VSCode and develop using a remote server. Works great. Couldn't get Ray working with it (tried some port forwarding and things) so ended up here looking for answers. 

Will keep an eye on future releases.

**Reply by Terrorboy** [reply]
@sunscreem 
Need urgently?
Try ngrok.

This method will remain in effect until formal support in the future.

https://github.com/spatie/ray/discussions/174#discussioncomment-340336

`ngrok http 23517`

**Reply by sunscreem** [reply]
Thanks @Terrorboy. No rush for me. Appreciate the help though

**Reply by marcorieser** [reply]
I adapted @Terrorboy's approach to use [expose from beyond code](https://beyondco.de/docs/expose/introduction):
* Install expose and run this in the terminal
`expose share http://localhost:23517`
* Insert the generated remote host and port 80 into the ray config
```php
return [
    ...
    'host' => 'xyz.sharedwithexpose.com',
    'port' => 80,
    ...
];
```

**Comment by sunscreem** [comment]
Just an FYI - I'm using the latest version and the remote server + debugging works a treat out of the box now.
